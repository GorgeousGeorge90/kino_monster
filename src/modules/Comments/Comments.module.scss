

.comments_container {
  @apply flex flex-col gap-2 items-center p-5;

  .comments_title {
    @apply font-bold first-letter:uppercase text-neutral-100 text-lg;
  }

  .comments_list {
    @apply flex flex-col gap-3;
  }

  .comments_null {
    @apply flex flex-col items-center text-neutral-100 text-lg cursor-pointer animate-open
    hover:before:content-[attr(data-hover)] hover:text-neutral-100 hover:text-sm;}

  .comments_limit {
    @apply first-letter:uppercase text-red-600 animate-open;
  }
}
