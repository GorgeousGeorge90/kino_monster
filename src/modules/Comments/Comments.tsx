import styles from './Comments.module.scss';
import { CommentType } from './types/types';
import man from './img/man.png';
import { FC } from 'react';


export type CommentsProps = {
    comments:CommentType[],
}

const Comments:FC<CommentsProps> =({comments}) => {

    return (<div className={styles.comments_container}>
        <div className={styles.comments_content}>
            {
                <ul className={styles.comments_list}>
                    {
                        comments.map(comment => <li className={styles.comments_list_item}
                                                    key={comment.id}>
                            <img className={styles.comments_list_item_logo}
                                src={man} alt={'pic'}/>
                            <p>{comment.comment}</p>
                        </li>)
                    }
                </ul>
            }
        </div>
    </div>)
}

export default Comments