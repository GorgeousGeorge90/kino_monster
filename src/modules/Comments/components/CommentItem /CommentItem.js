import styles from './CommentItem.module.scss';
import {useDispatch} from "react-redux";
import {deleteComment} from "../../store/commentsSlice";
import man from './../../img/man.png';

const CommentItem = ({name,comment,id}) => {
    const dispatch = useDispatch()

    return (<li className={styles.container}>
        <span className={styles.avatar}>
            <img src={man} alt="avatar"/>
        </span>
        <span>{name}:</span>
        <span>{comment}</span>
        <span className={styles.btn}
              onClick={()=>dispatch(deleteComment(id))}>X</span>
    </li>)
}

export default CommentItem