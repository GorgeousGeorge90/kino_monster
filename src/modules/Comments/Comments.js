import {useSelector} from 'react-redux';
import {getComments} from "./selectors/selectors";
import CommentItem from "./components/CommentItem /CommentItem";
import styles from './Comments.module.scss';


const Comments = () => {
    const comments = useSelector(state=>getComments(state))


    return (<div className={styles.container}>
        <ul>
            {
                comments.map(comment => <CommentItem key={comment.id}
                                                         {...comment}
                />)
            }
        </ul>
    </div>)
}

export default Comments