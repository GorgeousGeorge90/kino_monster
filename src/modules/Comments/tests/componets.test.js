import { getByRole, render, screen } from "@testing-library/react";
import CommentItem from "../components/CommentItem /CommentItem";
import {useSelector} from "react-redux";
import Comments from "../Comments";


jest.mock('react-redux')

describe('Comments components tests',()=>{
    it('goal: render CommentItem',()=> {

        const data = {
            id:'1',
            name:'Kate',
            comment: 'Hi!'
        }

        useSelector.mockReturnValue(data)

        const component = render(<CommentItem {...data}/>)
        const item = screen.getByRole('listitem')
        expect(item).toHaveTextContent('Kate')
        expect(component).toMatchSnapshot()
    })
})