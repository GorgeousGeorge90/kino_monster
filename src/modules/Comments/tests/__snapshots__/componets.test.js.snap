// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Comments components tests goal: render CommentItem 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <li
        class="container"
      >
        <span
          class="avatar"
        >
          <img
            alt="avatar"
            src="man.png"
          />
        </span>
        <span>
          Kate
          :
        </span>
        <span>
          Hi!
        </span>
        <span
          class="btn"
        >
          X
        </span>
      </li>
    </div>
  </body>,
  "container": <div>
    <li
      class="container"
    >
      <span
        class="avatar"
      >
        <img
          alt="avatar"
          src="man.png"
        />
      </span>
      <span>
        Kate
        :
      </span>
      <span>
        Hi!
      </span>
      <span
        class="btn"
      >
        X
      </span>
    </li>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Comments components tests goal: render Comments 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="container"
      >
        <form
          class="form"
        >
          <input
            placeholder="name"
            type="text"
            value=""
          />
          <input
            placeholder="city"
            type="text"
            value=""
          />
          <input
            placeholder="comment"
            type="text"
            value=""
          />
          <button>
            +
          </button>
        </form>
        <ul>
          <li
            class="container"
          >
            <span
              class="avatar"
            >
              <img
                alt="avatar"
                src="man.png"
              />
            </span>
            <span>
              Egor
              :
            </span>
            <span>
              Hi
            </span>
            <span
              class="btn"
            >
              X
            </span>
          </li>
        </ul>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="container"
    >
      <form
        class="form"
      >
        <input
          placeholder="name"
          type="text"
          value=""
        />
        <input
          placeholder="city"
          type="text"
          value=""
        />
        <input
          placeholder="comment"
          type="text"
          value=""
        />
        <button>
          +
        </button>
      </form>
      <ul>
        <li
          class="container"
        >
          <span
            class="avatar"
          >
            <img
              alt="avatar"
              src="man.png"
            />
          </span>
          <span>
            Egor
            :
          </span>
          <span>
            Hi
          </span>
          <span
            class="btn"
          >
            X
          </span>
        </li>
      </ul>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
